integration_tests/integration_test.go

```diff
@@ -94,8 +91,8 @@ func testGenericService(brokerConfig *config.BrokerConfig, gcpBroker *GCPService
 	Expect(err).ToNot(HaveOccurred())
 
 	// Provision is registered in the database
-	var count int
-	db_service.DbConnection.Model(&models.ServiceInstanceDetails{}).Where("id = ?", params.instanceId).Count(&count)
+	count, err := db_service.CountServiceInstanceDetailsById(params.instanceId)
```

Comment:
All of these CountXByYAndZ and CheckDeletedXByYAndZ could probably be rolled into a single "Exists" checker which would be a special case of "GetXByYAndZ", but that's for a later time.